inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -319,24 +334,52 @@ void SubstituteSoftSignPass::run() {
         }
         if (cont) continue;
         if (!hasNChildren(abs, 1)) continue;
-        auto power = getNthChild(abs, 0);
+        auto addition = getNthChild(abs, 0);
+        InferenceEngine::CNNLayerPtr power = nullptr;
+
+        if (!LayerInfo(addition).isPower()) continue;
+        auto powerLayer = LayerInfo(addition).as<PowerLayer*>();
+
+        if (!fp32eq(powerLayer->power, -1.0f)) {
+            //if first one is power equals 1, then it still can be a softsign
+            //with two power operations, one for add and one for power
+            if (!fp32eq(powerLayer->power, 1.0f) ||
+                !fp32eq(powerLayer->offset, 1.0f) ||
+                !fp32eq(powerLayer->scale, 1.0f)) {
+                continue;
+            } else {
+                power = getNthChild(addition, 0);
+                if (!LayerInfo(power).isPower()) continue;
+                auto powerLayer_1 = LayerInfo(power).as<PowerLayer*>();
+                if (!fp32eq(powerLayer_1->power, -1.0f) ||
+                    !fp32eq(powerLayer_1->offset, 0.0f) ||
+                        !fp32eq(powerLayer_1->scale, 1.0f)) continue;
+            }
+        } else {
+            if (!fp32eq(powerLayer->offset, 1) ||
+                !fp32eq(powerLayer->scale, 1)) continue;
 
-        if (!LayerInfo(power).isPower()) continue;
-        auto powerLayer = LayerInfo(power).as<PowerLayer*>();
-        if (powerLayer->power != -1) continue;
-        if (powerLayer->offset != 1) continue;
-        if (powerLayer->scale != 1) continue;
+            std::swap(addition, power);
+        }
```

Comment:
replace this nested if / else statemsents with:
```
// first layer after abs must have scale of 1, offset of 1 and power of either 1 or -1
if (!fp32eq(powerLayer.scale, 1.0f) || !fp32eq(powerLayer.offset, 1.0f) || !fp32eq(abs(powerLayer->power, 1.0f))) continue;
// power == -1, offset = 1, scale = 1
if(fp32eq(powerLayer->power, -1.0f)) {
    std::swap(addition, power);
} else { // power = 1, offset = 1, scale - 1 
    power = getNthChild(addition, 0);
    if (!LayerInfo(power).isPower()) continue;
    auto powerLayer_1 = LayerInfo(power).as<PowerLayer*>();
    // layer after addition must have power of -1, offset of 0 and scale of 1
    if (!(fp32eq(powerLayer_1->power, -1.0f) || !fp32eq(powerLayer_1->offset, 0.0f) || !fp32eq(powerLayer_1->scale, 1.0f)) continue;
}
```
The flow is much clearer this way and comments show exactly what to expect from each case